### ISSUE TITLE
[BUG] DataFrame `to_arrow(preserve_index=True)` doesn't preserve index in edge case

### ISSUE THREAD

**Issue by mj3cheun** [rapidsai/cudf#16682]
**Describe the bug**

When the method `to_arrow` is called on a cudf dataframe containing no columns but more than 1 row, with an index using the class `RangeIndex` with `preserve_index` set to `True`, the resulting pyarrow Table is empty.

By comparison, if the same is done with a cudf dataframe with an index using the class `Int64Index`, the resulting pyarrow Table contains an index column as expected.

**Steps/Code to reproduce bug**

Code to reproduce bug
```python
cudf.DataFrame(index=range(4)).to_arrow(preserve_index=True)
```
Output
```
pyarrow.Table

----
```

**Expected behavior**
```
pyarrow.Table
index: int64
----
index: [[0,1,2,3]]
```

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda
 - cuDF Version: 24.04.01

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**

**Comment by mroeschke** [comment]
Thanks for the report. Could you try this on the latest version of rapids (24.08)? I get the expected result there


```python
>>> import cudf
>>> cudf.DataFrame(index=range(4)).to_arrow(preserve_index=True)
pyarrow.Table
__index_level_0__: int64
----
__index_level_0__: [[0,1,2,3]]
>>> cudf.__version__
'24.08.02'
```

**Comment by mj3cheun** [comment]
same, looks like the problem is fixed on the latest version (24.08). thanks!
